### PR TITLE
feat/PS-96/sa_module

### PR DESCRIPTION
**Pull request by danperezsan**
Test: https://protraderseco.atlassian.net/browse/PS-96?focusedCommentId=10483

### CODE REVIEW

**Review by gbmeuk — CHANGES_REQUESTED** [review]
Automated tests, not comments.

**Comment by danperezsan** [comment]
Added unit and deployment tests

**Comment by danperezsan** [comment]
On the Setup Cloud SDK step for the deployment test the service account seems to need a new auth token:
```
Error: google-github-actions/setup-gcloud failed with: failed to execute command `gcloud --quiet auth activate-service-account githubactions-mcp-deploy@protean-buffer-230514.iam.gserviceaccount.com --key-file -`: ERROR: (gcloud.auth.activate-service-account) There was a problem refreshing your current auth tokens: ('invalid_grant: Invalid JWT Signature.', ***'error': 'invalid_grant', 'error_description': 'Invalid JWT Signature.'***)
```

File: tests/gcp/unit_tests/google_service_account/main.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+data external test_disabled {
```
Comment (gbmeuk) [inline tests/gcp/unit_tests/google_service_account/main.tf:1]: Can we test getting an IAM conditions field instead. That's a more complex algorithm.

File: tests/gcp/unit_tests/google_service_account/test_iam_bindings_count.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+except Exception as e:
+    print(e, stderr)
+
+
+"""
+    Checks the data specified in the required metadata
+"""
+
+expected_data = {
+    'test_account_count' : '1',
```
Comment (gbmeuk) [inline tests/gcp/unit_tests/google_service_account/test_iam_bindings_count.py]: Not useful to test the same value. Best practice is extremes and somewhere in the middle. I'd suggest null, 1 and greater than 1. Maybe test a members count as well.

File: Google/resource-manager/service_account/provider.tf [hunk 3]
```diff
@@ -0,0 +1,12 @@
+terraform {
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+      version = ">=3.55.0, <4.0.0"
+    }
+    google-beta = {
```
Comment (gbmeuk) [inline Google/resource-manager/service_account/provider.tf]: Are we using the beta provider?

File: tests/gcp/deployment/google_service_account/resource/service_accounts.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+project_id: <project-id>
+components:
+  specs:
+    test-account:
+      account_id: test-account
+      display_name: test-account
+      description: Test account
+      disabled: false
```
Comment (gbmeuk) [inline tests/gcp/deployment/google_service_account/resource/service_accounts.yaml:8]: Both of these SAs have disabled as false, so not a great test. I'd be inclined to have a service account for each test and name it accordingly. Makes it clearer and more robust.

**Review by gbmeuk — CHANGES_REQUESTED** [review]

**Review by gbmeuk — APPROVED** [review]

**Pull request merged**